features/FEATURE_BLE/source/generic/GenericGap.tpp

```diff
@@ -284,64 +212,47 @@ static bool is_random_private_resolvable_address(
         is_random_private_non_resolvable_address(address) ||
         is_random_static_address(address);
 }
 
-/*
- * Check disconnection reason validity.
- */
-static bool is_disconnection_reason_valid(::Gap::DisconnectionReason_t reason)
-{
-    switch (reason) {
-        /**
-         * Note: accepted reasons are:
-        typedef pal::disconnection_reason_t reason_t;
-        case reason_t::AUTHENTICATION_FAILLURE:
-        case reason_t::REMOTE_USER_TERMINATED_CONNECTION:
-        case reason_t::REMOTE_DEVICE_TERMINATED_CONNECTION_DUE_TO_LOW_RESOURCES:
-        case reason_t::REMOTE_DEVICE_TERMINATED_CONNECTION_DUE_TO_POWER_OFF:
-        case reason_t::UNSUPPORTED_REMOTE_FEATURE:
-        case reason_t::PAIRING_WITH_UNIT_KEY_NOT_SUPPORTED:
-        case reason_t::UNACCEPTABLE_CONNECTION_PARAMETERS:
-            */
-
-        // TODO Fix Disconnectionreason_t which expose invalid value
-        case ::Gap::REMOTE_USER_TERMINATED_CONNECTION:
-        case ::Gap::REMOTE_DEV_TERMINATION_DUE_TO_LOW_RESOURCES:
-        case ::Gap::REMOTE_DEV_TERMINATION_DUE_TO_POWER_OFF:
-        case ::Gap::CONN_INTERVAL_UNACCEPTABLE:
-            return true;
-        default:
-            return false;
-    }
-}
-
 /*
  * Return true if the whitelist in input is valid or false otherwise.
  */
-static bool is_whitelist_valid(const ::Gap::Whitelist_t &whitelist)
+static bool is_whitelist_valid(const ::ble::whitelist_t &whitelist)
 {
     if (whitelist.size > whitelist.capacity) {
         return false;
     }
 
     for (size_t i = 0; i < whitelist.size; ++i) {
-        const BLEProtocol::Address_t &address = whitelist.addresses[i];
-        if (address.type != BLEProtocol::AddressType::PUBLIC &&
-            address.type != BLEProtocol::AddressType::RANDOM_STATIC
-        ) {
-            return false;
-        }
-
-        if (address.type != BLEProtocol::AddressType::PUBLIC) {
-            if (is_random_address(address.address) == false) {
+        const whitelist_t::entry_t &entry = whitelist.addresses[i];
+        switch (entry.type.value()) {
+            case peer_address_type_t::PUBLIC:
+            case peer_address_type_t::PUBLIC_IDENTITY:
+                if (is_random_address(entry.address)) {
+                    return false; // invalid address bytes
+                }
+                break;
+            case peer_address_type_t::RANDOM:
+                if (!is_random_static_address(entry.address)
+                    && !is_random_private_resolvable_address(entry.address)) {
+                    // is either non-resolvable or contains invalid address bytes
+                    return false;
+                }
+                break;
+            case peer_address_type_t::RANDOM_STATIC_IDENTITY:
+                if (!is_random_static_address(entry.address)) {
+                    return false; // invalid address bytes
+                }
+                break;
+            default: // ANONYMOUS
                 return false;
```

Comment:
It's kind of reworked - the old and new address types do not match exactly...

Reply:
That's fair